src/main/java/org/graylog/plugins/threatintel/ThreatIntelPluginConfiguration.java

```diff
@@ -19,7 +19,8 @@
     public abstract boolean otxEnabled();
 
     @JsonProperty("otx_api_key")
-    public abstract Optional<String> otxApiKey();
+    @Nullable
+    public abstract String otxApiKey();
```

Comment:
Wouldn't this be enough so we can keep the nice properties of `java.util.Optional`?

```diff
diff --git src/main/java/org/graylog/plugins/threatintel/ThreatIntelPluginConfiguration.java src/main/java/org/graylog/plugins/threatintel/ThreatIntelPluginConfiguration.java
index 130fa5d..9b983fa 100644
--- src/main/java/org/graylog/plugins/threatintel/ThreatIntelPluginConfiguration.java
+++ src/main/java/org/graylog/plugins/threatintel/ThreatIntelPluginConfiguration.java
@@ -38,7 +37,7 @@ public abstract class ThreatIntelPluginConfiguration {
                                                         @JsonProperty("abusech_ransom_enabled") boolean abusechRansomEnabled) {
         return builder()
                 .otxEnabled(otxEnabled)
-                .otxApiKey(otxApiKey)
+                .otxApiKey(Optional.ofNullable(otxApiKey))
                 .torEnabled(torEnabled)
                 .spamhausEnabled(spamhausEnabled)
                 .abusechRansomEnabled(abusechRansomEnabled)
@@ -62,7 +61,7 @@ public abstract class ThreatIntelPluginConfiguration {
     public static abstract class Builder {
         public abstract Builder otxEnabled(boolean otxEnabled);
 
-        public abstract Builder otxApiKey(String otxApiKey);
+        public abstract Builder otxApiKey(Optional<String> otxApiKey);
 
         public abstract Builder torEnabled(boolean torEnabled);
```

Reply:
Maybe, but `Optional<T>` is pretty useless in this case anyway (check the usages of `ThreatIntelPluginConfiguration#otxApiKey()`).
In other words, getting rid of the `Optional<T>` meant having less code.

Reply:
I wouldn't call it useless, but fair enough.